### ISSUE TITLE
Calling extend on an ES EmberObject subclass does not call `constructor`

### ISSUE THREAD

**Issue by mitchlloyd** [emberjs/ember.js#14473]
As documented in https://github.com/emberjs/ember.js/pull/14462, calling `extend` on an ES subclass of Ember.Object yields a factory whose objects don't call `constructor` when constructed. 

Easier to explain with JS:

``` js
class SubEmberObject extends EmberObject {
  constructor() {
    // is never called
  }
}

let MyObject = SubEmberObject.extend({});
MyObject.create();
```

Note that the following does work:

``` js
class SubEmberObject extends EmberObject {
  constructor() {
    // is called
  }
}

class MyObject extends SubEmberObject {}
MyObject.create();
```

**Comment by pixelhandler** [comment]
@mitchlloyd I'm curious about `new` as well.

```
class SubEmberObject extends EmberObject {
  constructor() {
    // is this called?
  }
}
new SubEmberObject();
```

Typically with a class `new` is used. I find the this confusing without using `new`.

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes
- https://www.fourkitchens.com/blog/article/practical-introduction-es2015-classes

I know in Ember we're so used to `.create()` since Ember.Object makes that easy, otherwise we'd be doing this… https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create

**Comment by rwjblue** [comment]
The idea here (for the future world where we can use ES6 classes when they work for us) is that we can interchange `new` and `.create` (I.e. the consumer of the class doesn't have to care if it is a ES6 class or the result of Ember.Object.extend).

**Comment by pixelhandler** [comment]
@rwjblue yeah that is what I suspected, thanks for the clarity.

**Comment by stefanpenner** [comment]
I believe @krisselden has a fix in-mind.

**Comment by stefanpenner** [comment]
> can interchange new and .create

i don't know if we want total interchangeability here. For example, `create` has many legacy concepts (like most of [the following code](https://github.com/emberjs/ember.js/blob/master/packages/ember-runtime/lib/system/core_object.js#L75-L150) I would love for `new` to consider a 'new' start for some concepts :P

**Comment by pixelhandler** [comment]
@mitchlloyd @rwjblue @stefanpenner is this still an issue, perhaps we should close or create a new reproduction of this, what do you think? 

Now that we can actually use ES2015 classes maybe this should just be closed. https://medium.com/build-addepar/es-classes-in-ember-js-63e948e9d78e

**Comment by pixelhandler** [comment]
Per our triage policy I'll close this out for now, feel free to reopen if the issue persists on the current release.

- https://github.com/emberjs/ember.js/blob/master/CONTRIBUTING.md#triaging-policy

**Comment by rwjblue** [comment]
FWIW this was fixed in 3.2 or 3.3
